### PR TITLE
Hotfix/isbn canonicalization

### LINKED ISSUE #2473: Invalid isbns causing isbn10/13 issues in AffiliateLinks
This is a patron issue sent by email. "I believe this may of happened because I attempted to find further information about the book and when my edited work came up in search, I clicked on it .... rcvd the message below, then. I completed my file and clicked Send .... and once more, the same message popped out. "

### Relevant url?
https://openlibrary.org/books/OL27328216M/Gabriel's_Children

### Expectation
Viewing edition should not produce error.

### Details

This is from a patron, so details unknown.

```
/opt/openlibrary/deploys/openlibrary/fd32695/openlibrary/templates/type/edition/view.html: error in processing template: TypeError: 'NoneType' object is not iterable (falling back to default template)
```

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #2473

hotfix

### Technical

Removed scribd code which was in the way and being appropriated special cased to show up in the AffiliateLinks sidebar. Scribd still shows up under identifiers, yet no longer erroneously shows up as a buy option.

### Testing

Add an invalid `isbn_10` (e.g. `9198990604`) to an edition (which has no `isbn_13`). The live website edition page should throw the error below in https://github.com/internetarchive/openlibrary/pull/2475#issuecomment-539239357. Now it should load correctly. The invalid `isbn_10` should be shown correctly in the ID table. I believe an amazon link will appear (if you're on a kube w/ the right credentials).

### Evidence

https://openlibrary.org/books/OL27328216M/Gabriel's_Children -- if you look at the history, this page has only an `isbn_10` (`9198990604` -- which was invalid and) which should have been `0919899064` (since been fixed). The `isbn_10_to_isbn_13` was running `normalize_isbn` on the `isbn_10` which was causing an exception

### CODE REVIEW

**Comment by mekarpeles** [comment]
In openlibrary/utils/isbn.py, the problem lies in the fact that `isbn_10_to_isbn_13` will return `None` if the `isbn_10` is invalid and `canonical(None)` raises:

```
In [3]: canonical(None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-db6967b86bf2> in <module>()
----> 1 canonical(None)

/opt/openlibrary/venv/local/lib/python2.7/site-packages/isbnlib/_core.pyc in canonical(isbnlike)
    141 def canonical(isbnlike):
    142     """Keep only numbers and X."""
--> 143     numb = [c for c in isbnlike if c in '0123456789Xx']
    144     if numb and numb[-1] == 'x':
    145         numb[-1] = 'X'

TypeError: 'NoneType' object is not iterable
```

So we changed our `normalize_isbn` to `isbn and canonical(isbn) or None` and add a test for/where `isbn is None`

File: openlibrary/utils/tests/test_isbn.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ def test_opposite_isbn():
     assert opposite_isbn('BAD-ISBN') is None
 
 def test_normalize_isbn():
+    assert normalize_isbn(None) is None
```
Comment (mekarpeles) [inline openlibrary/utils/tests/test_isbn.py]: This is the business right here

**Comment by cdrini** [comment]
For future reference, let's avoid attaching arbitrary changes to hotfixes (the scribd change). That defeats the purpose of a hotfix, and slows down the whole process unnecessarily.
